### ISSUE TITLE
[FR] Metric aggregation options in experiment overview and comparison

### ISSUE THREAD

**Issue by CaptainTrojan** [mlflow/mlflow#7790]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

![image](https://user-images.githubusercontent.com/49991681/216582756-1f959031-1b19-4601-a40b-cd09c2ee18f1.png)

The "Metrics" columns consists of individual tracked metrics relevant to listed runs, such as "validation loss" for example. For each run, every metric is usually a list of numbers, as the value developed through time. However, for the overview, it is necessary to show only a single representative value. This, by default, is always the **last tracked value**, which cannot be changed.

This feature would allow users to specify exactly how should the list be aggregated into a single value using predefined operations from a dropdown menu (or something), such as **mean**, **max**, **min**, **median**, **last** and possibly more.

It would be highly beneficial to allow this in the "Compare" view as well, where run+metric is also represented by a single number, which is - again - the last tracked value.

### Motivation

> #### What is the use case for this feature?

A researcher wants to represent a run/metric pair with a different value than the last tracked, for example, when using early stopping with patience, the true value will not be the last one. Or perhaps they want to see the best value overall regardless of whether early stopping is used or not. The last tracked value is (imho) a very poor representation of the overall run/metric pair.

> #### Why is this use case valuable to support for MLflow users in general?

Because MLFlow users in general would like to see descriptive aggregations of their runs, not an uninteresting last tracked value. It would vastly reduce the time required for a user to invest in finding out which run is actualy performing well and which is not. My previous employee used a custom MLFlow logger that manually logged the best value at the end of every run just so that the overview contained this actually useful value and not the last tracked value as per default. 

> #### Why is this use case valuable to support for your project(s) or organization?

The use case is very non-specific to my projects. I'd just like to use MLFlow in the future and lack of this feature is a major hindrance in general.

> #### Why is it currently difficult to achieve this use case?

The UI does not support the selection of custom metric aggregation. An option is to implement a custom logger, which manually logs the expected value (mean, max, median, etc.) at the end of every run, but first, this doesn't allow later updates of the aggregation function, and second, it messes up the outcoming plots.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by dbczumar** [comment]
@CaptainTrojan Does having the ability to plot the train / val loss curves for multiple runs on the same axis help? We'll be improving this view and embedding it directly on the experiment page shortly.

**Comment by CaptainTrojan** [comment]
@dbczumar Very little. Yes, it is possible to find the best performing model that way, as the table in the compare/metric view shows you min and max values of said metric for each run in the selection, but you cannot compare using multiple metrics and all those scatter plot / parallel coordinate plot visualizations, which I was looking forward to using, are useless for me, because, well, _last tracked value_ is not representative. Not to mention that the overview metric columns are not representative as well.

**Comment by dbczumar** [comment]
@CaptainTrojan Got it - we'll be introducing a revamped experience for this shortly; you will be able to see multiple metric plots side by side and visualize them across time or epoch. We hope this will ease the burden of model selection. In the future, I completely agree that tracking the min / max / mean of each metric automatically would be very helpful. This will require some substantial changes to our REST API and as such isn't easily prioritized, but we've seen similar requests and will be mindful of them going forward. Thank you for using MLflow!

**Comment by CaptainTrojan** [comment]
Multiple plots at the same time sounds great. It's a shame that this feature can't be implemented yet though. Thanks for clarifying.
